File: 0706-design-hashmap/0706-design-hashmap.cs
public class MyHashMap
{
    private int capacity;
    private List<Bucket> buckets;

    public MyHashMap()
    {
        capacity = 11;
        buckets = new List<Bucket>(capacity);
        for (int i = 0; i < capacity; i++) 
        {
            buckets.Add(new Bucket());
        }
    }

    private int HashFunction(int key)
    {
        return Math.Abs(key) % capacity;
    }

    public void Put(int key, int value)
    {
        int hash = HashFunction(key);
        Bucket bucket = buckets[hash];
        bool keyFound = false;
        foreach (var tab in bucket.Data)
        {
            if (tab[0] == key)
            {
                tab[1] = value;
                keyFound = true;
                break;
            }
        }
        if (!keyFound)
        {
            bucket.Data.Add(new int[] { key, value });
        }
    }

    public int Get(int key)
    {
        int hash = HashFunction(key);
        Bucket bucket = buckets[hash];
        foreach (var tab in bucket.Data)
        {
            if (tab[0] == key)
            {
                return tab[1];
            }
        }
        return -1;
    }

    public void Remove(int key)
    {
        int hash = HashFunction(key);
        Bucket bucket = buckets[hash];
        int index = -1;
        bool keyFound = false;
        for (int i = 0; i < bucket.Data.Count; i++)
        {
            if (bucket.Data[i][0] == key)
            {
                index = i;
                keyFound = true;
                break;
            }
        }
        if (keyFound)
        {
            bucket.Data.RemoveAt(index);
        }
    }
}

public class Bucket
{
    public List<int[]> Data { get; }

    public Bucket()
    {
        Data = new List<int[]>();
    }
}


/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.Put(key,value);
 * int param_2 = obj.Get(key);
 * obj.Remove(key);
 */

File: 0706-design-hashmap/0706-design-hashmap.py
class MyHashMap:
    def __init__(self):
        self.capacity = 11
        self.buckets = [Bucket() for _ in range(self.capacity)]

    def hash_function(self, key):
        return abs(key) % self.capacity

    def put(self, key, value):
        hash = self.hash_function(key)
        bucket = self.buckets[hash]
        key_found = False
        for tab in bucket.data:
            if tab[0] == key:
                tab[1] = value
                key_found = True
                break
        if not key_found:
            bucket.data.append([key, value])

    def get(self, key):
        hash = self.hash_function(key)
        bucket = self.buckets[hash]
        for tab in bucket.data:
            if tab[0] == key:
                return tab[1]
        return -1

    def remove(self, key):
        hash = self.hash_function(key)
        bucket = self.buckets[hash]
        for i, pair in enumerate(bucket.data):
            if pair[0] == key:
                del bucket.data[i]
                break

class Bucket:
    def __init__(self):
        self.data = []


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)